### DISCUSSION TITLE
How do you navigate next and prev inside a collection of document in nextjs?

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#33284] [Help]
I'm not sure what its called, but I would like to be able to navigate between items inside a collection of posts(journals). For eg, If I am on an item (entry) I would like to be able to click next entry and prev entry in that collection(journal) .

Here is my site, which consists of journals(collection) and entries(items).
https://trip-journals-web.vercel.app/

I was thinking of somehow getting the ids of that collection and then using context to keep track of where I'm at, but it seems very complex. 

for eg. if I'm on this detail page...
https://trip-journals-web.vercel.app/entries/Oc8jw8QUCc5eKUt11WHc?journalId=c9b9y8m68FDrnIBTTEsI
I would like to either go to the prev page or next page. 

Right now, to do this, I have to go back to the summary page(the parent page):
https://trip-journals-web.vercel.app/journals/c9b9y8m68FDrnIBTTEsI

What is a good solution for this?

**Comment by TonyIngall** [comment]
Hey there!
Depends how you're storing your data, but I think the IDs might be along the right lines.
If trips are stored something like this:
```json
{
  "trips": [
    {
      "id": 1,
      "title": "Tokyo 2012",
      "tags": ["tokyo", "japan", "trip"],
      "events": [
        {
          "id": 1
          "title": "Sleeping overnight at a train station in the Winter"
        }
      ]
    }
  ]
  // etc, etc
}
```
Then I'm sure it wouldn't be too hard to extend it to do that. Since that's all stored in an array, what you would need to do from there is probably just loop through the array and check if it's the last or first item in the array to disable going forward or back any further respectively. Hopefully this helps in being able to at least think of a few ideas.
